Default.sublime-keymap

```diff
@@ -25,6 +25,24 @@
             }
         ]
     },
+    {
+        "command": "lsp_commit_completion_with_opposite_insert_mode",
+        "keys": [
+            "shift+enter"
```

Comment:
This also conflicts with the built-in key binding to insert a newline if the completion popup is visible.

It doesn't affect me personally, because I activated the `"auto_complete_commit_on_tab"` setting, but I would expect this to disturb many users if the built-in functionality suddenly isn't available anymore. It might be acceptable to provide key bindings which are direct replacements of built-in functionality, but with an override like this one here, I doubt that the LSP package would get accepted to Package Control if it had to re-apply for it.

Reply:
Can you point to the conflicting keybinding?

There are of `shift+enter` keybindings in ST, 
but none with a context when auto_complete_visible is visible.
```
            {
                "key": "auto_complete_visible",
                "operator": "equal",
                "operand": true
            },
```
So I did not see it conflicting.

Reply:
![Unbenannt](https://user-images.githubusercontent.com/6579999/188695585-45758c5e-06e5-4ec3-b4f8-05957cdcba4b.png)


The "context" only applys *restrictions* to a key binding. This one doesn't have a "context" restriction, so it is always activated. If the key binding from this PR gets added, it will be loaded after the default one, and therefore the default binding will be shadowed and won't work anymore.

Or do I have some misconception somewhere in my thoughts?

Reply:
Nope, you are correct. In this case the `lsp_commit_completion_with_opposite_insert_mode` keybinding will have precedence. :)

But is that bad? 
I mean other "shift+enter" ST keybinding also have contexts
and could also be seen as conflicting with this "shift+enter" insert `\n` keybinding, right?

In our case, 
the `lsp_commit_completion_with_opposite_insert_mode` will conflict only if the auto complete popup is visible and the user is in a file supported by LSP.

Reply:
Maybe this will make it more clear:
If you start ST in safe mode (all default settings), then you can commit a completion with <kbd>Enter</kbd>. But if you instead want to add a linebreak when the completion popup is visible, you can press <kbd>Shift</kbd>+<kbd>Enter</kbd>.
Now someone installs the LSP package and suddenly and without any warning <kbd>Shift</kbd>+<kbd>Enter</kbd> doesn't work anymore if a language server is running. Because it will then still commit the completion (with the opposite LSP insert mode), instead of the usual linebreak.

> But is that bad? I mean other "shift+enter" ST keybinding also have contexts and could also be seen as conflicting with this "shift+enter" insert `\n` keybinding, right?

But the other ST key bindings are for when the command palette or find or replace panel has focus. So they are not relevant regarding auto completions.
If we start overriding built-in key bindings for different functionality, then I guess we could also hijack for example <kbd>Ctrl</kbd>+<kbd>R</kbd> for ... let's say e.g. LSP's formatting feature. And if users want to keep using ST's "Goto Symbol" they can just reassign another key binding for it :D
Or maybe let's reassing ST's key binding for `toggle_comment` to something different, because who needs that anyway... :)

> In our case, the `lsp_commit_completion_with_opposite_insert_mode` will conflict only if the auto complete popup is visible and the user is in a file supported by LSP.

Well, but this is exactly the critical point. If the auto complete popup is not visible, then the built-in <kbd>Shift</kbd>+<kbd>Enter</kbd> binding is not useful anyway, because you can just press regular <kbd>Enter</kbd> without the <kbd>Shift</kbd> key to insert a linebreak.

Reply:
You explained that well, thanks!
OK, I will remove the keybinding, and let the user configure it :)

Reply:
[c6fac89](https://github.com/sublimelsp/LSP/pull/1809/commits/c6fac89ec0d718f6f799e948efcd78a1e7e7cf8f)